src/build-disk-image.ts

```diff
@@ -233,17 +233,39 @@ async function logContainer(image, containerId: string, progress, callback: (dat
   });
 }
 
+// find an unused container name
+export async function getUnusedName(name: string): Promise<string> {
+  let containers: string[] = [];
+  try {
+    // get a list of all existing container names, which may start with /
+    containers = (await extensionApi.containerEngine.listContainers())
+      .map(c => c.Names)
+      .reduce((a, val) => [...a, ...val], [])
+      .map(n => (n.charAt(0) === '/' ? n.substring(1) : n));
+  } catch (e) {
+    console.warn('Could not get existing container names');
+    console.warn(e);
+  }
+
+  let unusedName = name;
+  let count = 2;
+  while (containers.includes(unusedName)) {
+    unusedName = name + '-' + count++;
+  }
+  return unusedName;
+}
+
 // Create builder options for the "bootc-image-builder" container
-export function createBuilderImageOptions(
+export async function createBuilderImageOptions(
```

Comment:
I would prefer if we keep this non-async as all it does is generate the create options / not touching this function within this PR.

Instead, we should have getUnusedName outside of createBuilderImageOptions to get the name, and then simply pass it into createBuilderImageOptions.

Reply:
Yeah, that's better. Fixed in new commit.